src/cmap/connection_pool_events.ts

```diff
@@ -104,17 +127,21 @@ export class ConnectionClosedEvent extends ConnectionPoolMonitoringEvent {
   /** The reason the connection was closed */
   reason: string;
   serviceId?: ObjectId;
+  name: string = CONNECTION_CLOSED;
+  error: Error | undefined;
 
   /** @internal */
   constructor(
     pool: ConnectionPool,
     connection: Pick<Connection, 'id' | 'serviceId'>,
-    reason: string
+    reason: string,
+    error?: Error
   ) {
     super(pool);
     this.connectionId = connection.id;
     this.reason = reason || 'unknown';
```

Comment:
We have a switch statement based on this value now, so maybe lets improve this a bit, it can never be nullish so we can remove the `|| 'unknown'` case and can we make the type of reason **in the constructor** a union of string literals, we can leave the reason prop on the class as a string for now, but this improvement would help us not make a mistake internally